File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.py
def solution(s):
    s_upper = s.upper()
    x_count = 0
    y_count = 0
    
    for s_upper in s_upper: 
        if s_upper == 'P' :
            x_count+=1
        elif s_upper == "Y":
            y_count+=1
            
    if x_count == y_count :
        return True
    elif x_count != y_count: 
        return False

    

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        s = s.toLowerCase();
        
        int p_count = 0;
        int y_count = 0;
        
        for(int i=0; i<s.length(); i++){         
            for(char c : s.toCharArray()){                
                if(c == 'p'){
                    p_count++;
                }else if(c == 'y'){
                    y_count++;
                }
            }
            
        }
        
        if(p_count != y_count){
            answer = false;
        }

        return answer;
    }
}